platforms/core-configuration/input-tracking/src/main/java/org/gradle/internal/configuration/inputs/AccessTrackingProperties.java

```diff
@@ -388,12 +397,14 @@ public void clear() {
     }
 
     @Override
+    @Nullable
     public String getProperty(String key) {
         return getProperty(key, null);
     }
 
     @Override
-    public String getProperty(String key, String defaultValue) {
+    @Nullable  // The contract here is trickier - the return value is only null if the default value is null, but this methog is not really part of any public API.
```

Comment:
💅 There is a typo in word `methog`

💭 Is it possible to add IntelliJ's `Contract` annotation that expresses this nullability contract? IIUC, they support conditionals like `null -> null` and `non-null -> non-null` for parameter-return-value relationship.
We don't have to do this in this PR, but your comment triggered this idea.

Reply:
Thanks!

Regarding `@Contract`, I don't want to be a pioneer here :) The type itself isn't intended to be used directly, but rather through its base class, so it adds little value.